pulsar-client-admin/src/main/java/org/apache/pulsar/client/admin/Topics.java

```diff
@@ -1604,7 +1604,15 @@ void setDelayedDeliveryPolicy(String topic
      * @throws PulsarAdminException
      *             Unexpected error
      */
-    int getMessageTTL(String topic) throws PulsarAdminException;
+    Integer getMessageTTL(String topic) throws PulsarAdminException;
```

Comment:
this is a breaking change in the API, 
code compiled for previous versions of Pulsar client will have to be re-compiled.
And also this is a behaviour change, recompiling is not enough to apply the new meaning of the value.
People who simply recompile will see NPE
 
I would not change this method and make it that returns the same value as before.
we are adding the new method and that will be the right way.
Here we have to add a javadoc that explains the meaning of the result,
we could also deprecate this method

Reply:
Methods with the same parameters and names cannot coexist. This is a problem left over from history. If we want to switch completely smoothly, we can add interfaces like admin.topicsV2(), admin.namespaceV2()...

If we modify the interface directly, users will know when upgrading the SDK(It will definitely trigger a recompilation when upgrading the SDK), we need to improve the doc of each interface, clarify the meaning of each return value

Please take a look here @sijie @codelipenghui

Reply:
@eolivelli This method is introduced in 2.7.0(topic level policy) and I think this is a mistake in 2.7.0, most of the `get policy` method are returned Integer or Long, and the old behavior is the wrong behavior. I think we can clarify it in the release doc, I prefer to fix it directly. If left this method for future releases, users might be confused more.

Reply:
We must pay more attention to public APIs, this kind of changes will hurt users, probably someone will get an unexpected NPE

I would accept to change the signature for a 2.8. release, but please do not cherry pick this change to 2.7 branch.

Reply:
topic policy is a newer feature that is not advertised as a stable change yet in 2.7.0. I think we should fix the bad behavior in the 2.7.x release.

Reply:
It is different from breaking a change that is introduced in old versions than 2.7.x.

Reply:
I see.
let's fix it for 2.7.x